### DISCUSSION TITLE
Frigate NVR App for android wear

### DISCUSSION THREAD

**Discussion by algoz098** [blakeblackshear/frigate#10234] [Show and tell]
I have made a repo with the source code for a android wear app to use with frigate nvr.

I believe few ppl would want this, but as I was one, I have built, and been using it.
https://github.com/algoz098/frigate_wear

Any PRs or usage is welcome. Did not publish yet because I'm testing.

It can:
- internal URL which is used while in defined Networks;
- External URL which is used in any other case;
- Internal Networks, just a list of WIFI SSIDs for the internal url.
- It will show the cameras you have, when configured.
- It will show the JSMPeg like expirience to the selected camera.

Why tho?
Because sometime I want to keep a Eye to some camera while doing something else, like waching TV.

How?
It's flutter, just a set interval to refresh the image 1/s.

Help wanted to do it, I'm not a good flutter dev, so, was fun, but almost At my limit.
My frigate external link is behind cloudflared zero trust, so I could not use it in there.

**Comment by lkaino** [comment]
Awesome! I'm going to test it out. My external URL is behind google authentication, I guess that won't work.

**Comment by lkaino** [comment]
I'm not familiar with Flutter development, how do I run this on my Wear OS 3 watch?

**Reply by lkaino** [reply]
Okay I got it to work. Steps roughly for ubuntu:

1. install flutter with snap
2. install android studio from their website
3. unpack the tar and chmod +x bin/studio.sh
4. run it and install command line tools
5. build the app with flutter build apk
6. sideload to watch

**Reply by algoz098** [reply]
You have to be able to use flutter and android studio: https://docs.flutter.dev/get-started/install
After you have both working, you have to enable dev mode in your android wear: https://developer.android.com/training/wearables/get-started/debugging?hl=pt-br

Then you just have to compile it: 
- 'flutter pub get'
- 'flutter build apk'

Finaly to install it on the device: flutter install

There's a lot which can go wrong in the between, because of dev envirioments, PATH variables things out of place, so be mindfull to look in the why something is not working.

**Comment by lkaino** [comment]
It works and it's awesome. It would be nice to be able to scroll up/down between the cameras.
